{{cookiecutter.project_slug}}/frontend/app/lib/slices/authSlice.ts

```diff
@@ -109,54 +113,58 @@ export const login =
           icon: "error",
         }),
       )
+      dispatch(logout())
     }
   }
 
+export const login =
+  (payload: { username: string; password?: string }) => handleGenericLogin(getTokens, payload)
+
 export const magicLogin =
   (payload: { token: string }) =>
-  async (
-    dispatch: ThunkDispatch<any, void, Action>,
-    getState: () => RootState,
-  ) => {
-    try {
-      await dispatch(validateMagicTokens(payload.token))
-      const token = getState().tokens.access_token
-      await dispatch(getUserProfile(token))
-    } catch (error) {
-      dispatch(
-        addNotice({
-          title: "Login error",
-          content:
-            "Please check your details or internet connection and try again.",
-          icon: "error",
-        }),
-      )
-      dispatch(logout())
+    async (
+      dispatch: ThunkDispatch<any, void, Action>,
+      getState: () => RootState,
+    ) => {
+      try {
+        await dispatch(validateMagicTokens(payload.token))
+        const token = getState().tokens.access_token
+        await dispatch(getUserProfile(token))
+      } catch (error) {
+        dispatch(
+          addNotice({
+            title: "Login error",
+            content:
+              "Please check your details or internet connection and try again.",
+            icon: "error",
+          }),
+        )
+        dispatch(logout())
+      }
     }
-  }
 
 export const totpLogin =
   (payload: { claim: string }) =>
-  async (
-    dispatch: ThunkDispatch<any, void, Action>,
-    getState: () => RootState,
-  ) => {
-    try {
-      await dispatch(validateTOTPClaim(payload.claim))
-      const token = getState().tokens.access_token
-      await dispatch(getUserProfile(token))
-    } catch (error) {
-      dispatch(
-        addNotice({
-          title: "Login error",
-          content:
-            "Please check your details or internet connection and try again.",
-          icon: "error",
-        }),
-      )
-      dispatch(logout())
+    async (
```

Comment:
Same here.

Reply:
Yeah, these are just line-number changes. I'm actually going to curry them into the `handleGenericLogin` function, so they'll be gone as well.